plugins/woocommerce/src/Internal/DataStores/StockNotifications/StockNotificationsDataStore.php

```diff
@@ -0,0 +1,495 @@
+		if ( $changes ) {
+			$notification->apply_changes();
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Delete a stock notification.
+	 *
+	 * @param Notification $notification The data object to delete.
+	 * @param array        $args         Additional arguments.
+	 * @return void
+	 */
+	public function delete( &$notification, $args = array() ) {
+		global $wpdb;
+
+		$deleted = $wpdb->delete( $this->get_table_name(), array( 'id' => $notification->get_id() ), array( '%d' ) );
+
+		if ( is_int( $deleted ) && $deleted > 0 ) {
+			$this->data_store_meta->delete_by_notification_id( $notification->get_id() );
+		}
+	}
+
+	/**
+	 * Add meta.
+	 *
+	 * @param Notification $notification The data object to add.
+	 * @param \stdClass    $meta         The meta object to add (containing ->key and ->value).
+	 * @return int|false The meta ID or false if the meta was not added.
+	 */
+	public function add_meta( &$notification, $meta ) {
+		$add_meta = $this->data_store_meta->add_meta( $notification, $meta );
+		$this->after_meta_change( $notification );
+		return $add_meta ? $add_meta : false;
+	}
+
+	/**
+	 * Read meta.
+	 *
+	 * @param Notification $notification The data object to read.
+	 * @return array
+	 */
+	public function read_meta( &$notification ): array {
+		$raw_meta_data = $this->data_store_meta->read_meta( $notification );
+		return $this->filter_raw_meta_data( $notification, $raw_meta_data );
+	}
+
+	/**
+	 * Update meta.
+	 *
+	 * @param Notification $notification The data object to update.
+	 * @param \stdClass    $meta         The meta object to update (containing ->id, ->key and ->value).
+	 * @return bool
+	 */
+	public function update_meta( &$notification, $meta ): bool {
+		$update_meta = $this->data_store_meta->update_meta( $notification, $meta );
+		$this->after_meta_change( $notification );
+		return $update_meta;
+	}
+
+	/**
+	 * Delete meta.
+	 *
+	 * @param Notification $notification The data object to delete.
+	 * @param \stdClass    $meta         The meta object to delete (containing at least ->id).
+	 * @return bool
+	 */
+	public function delete_meta( &$notification, $meta ): bool {
+		$delete_meta = $this->data_store_meta->delete_meta( $notification, $meta );
+
+		$this->after_meta_change( $notification );
+		return $delete_meta;
+	}
+
+	/**
+	 * Perform after meta change operations.
+	 *
+	 * @param Notification $notification The notification object.
+	 * @return bool True if changes were applied, false otherwise.
+	 */
+	private function after_meta_change( &$notification ): bool {
+
+		$current_time      = time();
+		$current_date_time = new \WC_DateTime( "@$current_time", new \DateTimeZone( 'UTC' ) );
+
+		$should_save =
+			$notification->get_id() > 0
+			&& $notification->get_date_modified( 'edit' ) < $current_date_time
+			&& empty( $notification->get_changes() );
+
+		if ( $should_save ) {
+			$notification->set_date_modified( $current_time );
+			$saved = $notification->save();
+			return is_int( $saved ) && $saved > 0;
```

Comment:
With the latest changes, it seems to be a change in behaviour. WDYT about `return ! is_wp_error( $saved );`?

Reply:
Neat! Done in https://github.com/woocommerce/woocommerce/pull/57062/commits/4b66f897d2f7b514744f5d72e0ba9ee9b0ae2587